### ISSUE TITLE
Backward difference encoding changes DataFrame column names

### ISSUE THREAD

**Issue by kdmcdrm** [scikit-learn-contrib/categorical-encoding#125]
This might be a quick one as it looks like it's intentional, I just don't understand why. I tried a BackwardDifferenceEncoder and found that it prepended all of my columns with "col_". Digging into the library I found that is done on backward_difference, line 195.

Is there a reason all columns get renamed, including those that are not transformed by the encoder? I am going to add a line to my pipeline to rename them back to their original names, but this was unexpected behavior and I haven't been able to find any documentation on why it works this way.

**Comment by janmotl** [comment]
This behaviour can also be observed in HelmertEncoder, SumEncoder and PolynomialEncoder:
```Python
    def test_preserve_names(self):
        binary_cat_example = pd.DataFrame(
            {'ignore': ['UP', 'UP', 'DOWN', 'FLAT'],
             'feature': ['UP', 'UP', 'DOWN', 'FLAT'],
             'target': [1, 1, 0, 0]}, columns=['ignore', 'feature', 'target'])

        for encoder_name in encoders.__all__:
            with self.subTest(encoder_name=encoder_name):
                encoder = getattr(encoders, encoder_name)(cols=['feature'])
                result = encoder.fit_transform(binary_cat_example, binary_cat_example['target'])
                columns = result.columns.values

                self.assertTrue('ignore' in columns, "Column 'ignore' is missing in: " + str(columns))
```

**Comment by janmotl** [comment]
Solved with #128
